### PR TITLE
DEV-12137: Implement `renderOneLine` custom string literal template tag

### PR DESCRIPTION
**Pull request by anderspollack**
- Implement `renderOneLine` custom common-tag
- Use `renderOneLine` in `cite.js` shortcode
- Add content to `transforms` directory `README.md`

Creates a custom string literal template tag to handle rendering string literals on a single line without the new-line character replaced by space character.

### CODE REVIEW

**Comment by anderspollack** [comment]
@mphstudios I put this in `_plugins/transforms` since that was empty, and this seemed like a "transform" but I'm fine with renaming the directory something like "stringLiteralTemplateTags" or something

File: packages/11ty/_plugins/transforms/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+</span>
+```
+
+Gets rendered like this:
+
+```html
+<span> Some inline text </span>
+```
+
+Instead, `renderOneLine` renders it like this:
```
Comment (mphstudios) [inline packages/11ty/_plugins/transforms/README.md:23]:
```suggestion
Instead, `renderOneLine` will output the following string:
```

File: packages/11ty/_plugins/transforms/renderOneLine.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const { createTag, stripIndents } = require('common-tags')
+
+module.exports = createTag({
+  onString(string) {
+    return stripIndents(string)
+  },
+  onEndResult(string) {
+    return string.replace(/\n/g, '')
+  }
+});
```
Comment (mphstudios) [inline packages/11ty/_plugins/transforms/renderOneLine.js:10]:
```suggestion
})

```

**Review by mphstudios — APPROVED** [review]

File: packages/11ty/_plugins/common-tags/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+When using `oneLine`, a structure like this
+
+```html
+<span>
+  Some inline text
+</span>
+```
+
+Gets rendered like this:
```
Comment (mphstudios) [inline packages/11ty/_plugins/common-tags/README.md:17]:
```suggestion
replaces new-line characters with a space, resulting in unwanted white-space in the text.
```

File: packages/11ty/_plugins/common-tags/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+This directory includes templating tags to extend functions included by `common-tags`. **Nota bene**: these are not strictly 11ty plugins, but they are used by javascript component shortcodes.
+
+## `renderOneLine`
+
+Similar to `oneLine`, this function flattens html tags into a single line, stripping out whitespace that surrounds text nodes inside of tags. It allows template HTML for elements to include indentation for better readability without rendering the extra whitespace.
+
+### Example
+
+When using `oneLine`, a structure like this
```
Comment (mphstudios) [inline packages/11ty/_plugins/common-tags/README.md:9]:
```suggestion
Using `oneLine` on a multi-line string literal like the following,
```

**Pull request merged**